setup.sh

```diff
@@ -1,8 +1,10 @@
 #!/bin/sh
 
-
-#echo "$@"
+(
+cd $(dirname ${BASH_SOURCE[0]})
```

Comment:
${BASH_SOURCE[0]} only works in a script.  It gives the name of the script you are calling (whether from ./setup.sh or source ./setup.sh).